lint/eslint.bzl

```diff
@@ -63,13 +63,25 @@ def _gather_inputs(ctx, srcs):
     inputs = copy_files_to_bin_actions(ctx, srcs)
 
     # Add the config file along with any deps it has on npm packages
-    inputs.extend(js_lib_helpers.gather_files_from_js_providers(
-        ctx.attr._config_files + [ctx.attr._workaround_17660, ctx.attr._formatter],
-        include_transitive_sources = True,
-        include_declarations = False,
-        include_npm_linked_packages = True,
-    ).to_list())
-
+    if "gather_files_from_js_providers" in dir(js_lib_helpers):
+        # rules_js 1.x
+        js_inputs = js_lib_helpers.gather_files_from_js_providers(
+            ctx.attr._config_files + [ctx.attr._workaround_17660, ctx.attr._formatter],
+            include_transitive_sources = True,
+            include_declarations = False,
+            include_npm_linked_packages = True,
+        )
+    else:
+        # rules_js 2.x
+        js_inputs = js_lib_helpers.gather_files_from_js_infos(
```

Comment:
I don't expect this API to change between rules_js 2.0.0-alpha.4 and final so this PR is safe to land... tho maybe we should wait until RC so we don't have the public example point to alpha releases of rules_js & rules_ts

Reply:
Gotta land soon so we can try rjs2 downstream